kube_hunter/modules/discovery/hosts.py

```diff
@@ -61,12 +61,33 @@ def __init__(self, pod=False, active=False, predefined_hosts=None):
 class HostDiscoveryHelpers:
     # generator, generating a subnet by given a cidr
     @staticmethod
-    def generate_subnet(ip, sn="24"):
+    def generate_subnet(ip, sn="24", ignore=None):
         logger.debug(f"HostDiscoveryHelpers.generate_subnet {ip}/{sn}")
         subnet = f"{ip}/{sn}"
         for ip in IPNetwork(subnet):
-            logger.debug(f"HostDiscoveryHelpers.generate_subnet yielding {ip}")
-            yield ip
+            if ignore and any(ip in s for s in ignore):
+                logger.debug(f"HostDiscoveryHelpers.generate_subnet DENIED {ip}")
+            else:
+                logger.debug(f"HostDiscoveryHelpers.generate_subnet yielding {ip}")
+                yield ip
+
+    @staticmethod
+    def gen_ips():
```

Comment:
Take cidr as paramter rather than global config.
It will make this code more testable.